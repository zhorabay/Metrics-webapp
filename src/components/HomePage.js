import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSuperpowers } from '@fortawesome/free-brands-svg-icons';
import {
  faAngleRight, faVideo, faGamepad, faBolt,
} from '@fortawesome/free-solid-svg-icons';
import '../styles/HomePage.css';

const HomePage = () => {
  return (
    <section className="mainpage">
      <div className="mainpage-body">
        <ul className="mainpage-icons">
          <li className="icons-list">
            <a className="characters-link" href="characters">
              <FontAwesomeIcon className="in" icon={faAngleRight} />
              <FontAwesomeIcon className="icon" icon={faSuperpowers} />
              <div className="firstline">
                <h2 className="mainpage-title" to="/characters">CHARACTERS</h2>
                <p className="item-amount">1</p>
              </div>
            </a>
          </li>
          <li className="icons-list">
            <FontAwesomeIcon className="in" icon={faAngleRight} />
            <FontAwesomeIcon className="icon" icon={faBolt} />
            <div className="firstline">
              <h2 className="mainpage-title">COMICS</h2>
              <p className="item-amount">2</p>
            </div>
          </li>
          <li className="icons-list">
            <FontAwesomeIcon className="in" icon={faAngleRight} />
            <FontAwesomeIcon className="icon" icon={faGamepad} />
            <div className="firstline">
              <h2 className="mainpage-title">GAMES</h2>
              <p className="item-amount">3</p>
            </div>
          </li>
          <li className="icons-list">
            <FontAwesomeIcon className="in" icon={faAngleRight} />
            <FontAwesomeIcon className="icon" icon={faVideo} />
            <div className="firstline">
              <h2 className="mainpage-title">VIDEOS</h2>
              <p className="item-amount">4</p>
            </div>
          </li>
        </ul>
      </div>
      <footer>
        <a className="footer-link" href="http://marvel.com">Data provided by Marvel. Â© 2023 MARVEL</a>
      </footer>
    </section>
  );
};

export default HomePage;
