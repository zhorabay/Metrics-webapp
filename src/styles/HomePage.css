.mainpage {
  text-align: center;
}

.mainpage-body {
  background-color: red;
  display: flex;
  justify-content: center;
  align-items: center;
}

.mainpage-icons {
  display: grid;
  grid-template-columns: 250px 250px 250px 250px;
  grid-template-rows: 250px;
  gap: 10px;
  list-style: none;
  margin: 210px auto;
}

.icons-list {
  background-color: rgba(255, 255, 255, 0.4);
  padding: 15px;
  display: flex;
  flex-direction: column;
}

.icon {
  height: 90px;
  color: white;
  margin: 10px 0;
}

.firstline {
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.item-amount {
  font-size: 16px;
  color: white;
  margin: 0;
}

.mainpage-title {
  color: white;
  margin: 4px;
}

.in {
  color: white;
  border: 2px solid white;
  border-radius: 100px;
  padding: 10px 12px;
  margin-top: 6px;
  align-self: flex-end;
}

footer {
  background-color: rgb(50, 50, 50);
  padding: 18px;
}

.footer-link {
  text-decoration: none;
  color: white;
}

.characters-link {
  text-decoration: none;
  display: flex;
  flex-direction: column;
}
