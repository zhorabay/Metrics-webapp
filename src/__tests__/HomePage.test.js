import React from 'react';
import { render } from '@testing-library/react';
import HomePage from '../components/HomePage';

test('HomePage renders correctly', () => {
  const { container } = render(<HomePage />);
  expect(container).toMatchSnapshot();
});
