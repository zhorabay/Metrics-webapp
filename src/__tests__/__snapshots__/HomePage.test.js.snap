// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders correctly 1`] = `
<div>
  <section
    class="mainpage"
  >
    <div
      class="mainpage-body"
    >
      <ul
        class="mainpage-icons"
      >
        <li
          class="icons-list"
        >
          <a
            class="characters-link"
            href="characters"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-right in"
              data-icon="angle-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M278.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-160 160c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L210.7 256 73.4 118.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l160 160z"
                fill="currentColor"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-superpowers icon"
              data-icon="superpowers"
              data-prefix="fab"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M448 32c-83.3 11-166.8 22-250 33-92 12.5-163.3 86.7-169 180-3.3 55.5 18 109.5 57.8 148.2L0 480c83.3-11 166.5-22 249.8-33 91.8-12.5 163.3-86.8 168.7-179.8 3.5-55.5-18-109.5-57.7-148.2L448 32zm-79.7 232.3c-4.2 79.5-74 139.2-152.8 134.5-79.5-4.7-140.7-71-136.3-151 4.5-79.2 74.3-139.3 153-134.5 79.3 4.7 140.5 71 136.1 151z"
                fill="currentColor"
              />
            </svg>
            <div
              class="firstline"
            >
              <h2
                class="mainpage-title"
                to="/characters"
              >
                CHARACTERS
              </h2>
              <p
                class="item-amount"
              >
                1
              </p>
            </div>
          </a>
        </li>
        <li
          class="icons-list"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-angle-right in"
            data-icon="angle-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M278.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-160 160c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L210.7 256 73.4 118.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l160 160z"
              fill="currentColor"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-bolt icon"
            data-icon="bolt"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M349.4 44.6c5.9-13.7 1.5-29.7-10.6-38.5s-28.6-8-39.9 1.8l-256 224c-10 8.8-13.6 22.9-8.9 35.3S50.7 288 64 288H175.5L98.6 467.4c-5.9 13.7-1.5 29.7 10.6 38.5s28.6 8 39.9-1.8l256-224c10-8.8 13.6-22.9 8.9-35.3s-16.6-20.7-30-20.7H272.5L349.4 44.6z"
              fill="currentColor"
            />
          </svg>
          <div
            class="firstline"
          >
            <h2
              class="mainpage-title"
            >
              COMICS
            </h2>
            <p
              class="item-amount"
            >
              2
            </p>
          </div>
        </li>
        <li
          class="icons-list"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-angle-right in"
            data-icon="angle-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M278.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-160 160c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L210.7 256 73.4 118.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l160 160z"
              fill="currentColor"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-gamepad icon"
            data-icon="gamepad"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M192 64C86 64 0 150 0 256S86 448 192 448H448c106 0 192-86 192-192s-86-192-192-192H192zM496 168a40 40 0 1 1 0 80 40 40 0 1 1 0-80zM392 304a40 40 0 1 1 80 0 40 40 0 1 1 -80 0zM168 200c0-13.3 10.7-24 24-24s24 10.7 24 24v32h32c13.3 0 24 10.7 24 24s-10.7 24-24 24H216v32c0 13.3-10.7 24-24 24s-24-10.7-24-24V280H136c-13.3 0-24-10.7-24-24s10.7-24 24-24h32V200z"
              fill="currentColor"
            />
          </svg>
          <div
            class="firstline"
          >
            <h2
              class="mainpage-title"
            >
              GAMES
            </h2>
            <p
              class="item-amount"
            >
              3
            </p>
          </div>
        </li>
        <li
          class="icons-list"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-angle-right in"
            data-icon="angle-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M278.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-160 160c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L210.7 256 73.4 118.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l160 160z"
              fill="currentColor"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-video icon"
            data-icon="video"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 128C0 92.7 28.7 64 64 64H320c35.3 0 64 28.7 64 64V384c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V128zM559.1 99.8c10.4 5.6 16.9 16.4 16.9 28.2V384c0 11.8-6.5 22.6-16.9 28.2s-23 5-32.9-1.6l-96-64L416 337.1V320 192 174.9l14.2-9.5 96-64c9.8-6.5 22.4-7.2 32.9-1.6z"
              fill="currentColor"
            />
          </svg>
          <div
            class="firstline"
          >
            <h2
              class="mainpage-title"
            >
              VIDEOS
            </h2>
            <p
              class="item-amount"
            >
              4
            </p>
          </div>
        </li>
      </ul>
    </div>
    <footer>
      <a
        class="footer-link"
        href="http://marvel.com"
      >
        Data provided by Marvel. Â© 2023 MARVEL
      </a>
    </footer>
  </section>
</div>
`;
